deeplake/integrations/mmdet/mmdet_.py

```diff
@@ -1307,6 +1308,15 @@ def _train_detector(
     auto_scale_lr(cfg, distributed, logger)
     optimizer = build_optimizer(model, cfg.optimizer)
 
+    cfg.custom_imports = dict(
+        imports=["deeplake.integrations.mmdet.mmdet_runners"],
+        allow_failed_imports=False,
+    )
+    if cfg.runner.type == "IterBasedRunner":
+        cfg.runner.type = "DeeplakeIterBasedRunner"
+    elif cfg.runner.type == "EpochBasedRunner":
+        cfg.runner.type = "DeeplakeEpochBasedRunner"
```

Comment:
MMdet supports only two types of runners, IterBasedRunner and EpochBasedRunner. Putting elif just in case someone will want to use their own custom implemented runner